server/controllers/medical/patient.js

```diff
@@ -429,3 +452,148 @@ function logVisit(patientData, userId) {
 function isEmpty(array) {
   return array.length === 0;
 }
+
+/**
+* GET /patient/search?name="..."&details=...&limit=...
+*
+* @desc This function is responsible to find a patient with detailled informations or not
+* and with a limited rows or not
+*/
+function search (req, res, next) {
+
+  var sql,
+      data       = [],
+      qReference = req.query.reference,
+      qName      = req.query.name,
+      qFields    = req.query.fields,
+      qDetail    = req.query.detail || 0,
+      qLimit     = req.query.limit;
+
+  if (!qReference && !qName && !qFields) { throw req.codes.ERR_PARAMETERS_REQUIRED; }
```

Comment:
To increase code clarity, it would be better to use a local variable to store this boolean.  Like this:
```js
var hasRequiredParams = (!qReference && !qName && !qFields);
if (!hasRequiredParams) { throw req.codes.ERR_PARAMETERS_REQUIRED; }
```